Source/WebCore/Modules/url-pattern/URLPatternCanonical.cpp

```diff
@@ -176,13 +176,24 @@ ExceptionOr<String> canonicalizePort(StringView portValue, const std::optional<S
 // https://urlpattern.spec.whatwg.org/#canonicalize-an-opaque-pathname
 ExceptionOr<String> canonicalizeOpaquePathname(StringView value)
 {
+    if (value.isEmpty())
+        return value.toString();
+
+    // FIXME: Copy approach to remove prepending slash from canonicalizeOpaquePathname output. See note in https://urlpattern.spec.whatwg.org/#canonicalize-a-pathname. This function is now identical to the canonicalizePathname function below.
```

Comment:
Some of the test cases were failing because there was a prepended slash. So I had to add this.

Reply:
Nit: We could add FIXME comment like `// FIXME: Set state override to opaque path state after URLParser supports state override.` in this function and `canonicalizePathname`. Then we don't need to mention they are identical.